packages/nx/src/native/workspace/get_nx_workspace_files.rs

```diff
@@ -169,22 +172,23 @@ fn read_project_configuration(
     })
 }
 
-type WorkspaceData = (Vec<(String, Vec<u8>)>, Vec<FileData>);
+type WorkspaceData = (HashMap<PathBuf, Vec<u8>>, Vec<FileData>);
 fn get_file_data(workspace_root: &str, globs: Vec<String>) -> anyhow::Result<WorkspaceData> {
     let globs = build_glob_set(globs)?;
     let (projects, file_data) = nx_walker(workspace_root, move |rec| {
-        let mut projects: Vec<(String, Vec<u8>)> = vec![];
+        let mut projects: HashMap<PathBuf, (PathBuf, Vec<u8>)> = HashMap::new();
         let mut file_hashes: Vec<FileData> = vec![];
         for (path, content) in rec {
             file_hashes.push(FileData {
-                file: path.clone(),
+                file: path.to_normalized_string(),
                 hash: xxh3::xxh3_64(&content).to_string(),
             });
-            if globs.is_match(&path) {
-                projects.push((path, content));
-            }
+            insert_config_file_into_map((path, content), &mut projects, &globs)
         }
         (projects, file_hashes)
     });
-    Ok((projects, file_data))
+    Ok((
+        projects.into_iter().map(|(_, config)| config).collect(),
```

Comment:
There's a clippy suggestion for this:
```suggestion
        projects.into_values().collect(),
```